service/middleware/trace.go

```diff
@@ -45,26 +45,34 @@ func (l *Trace) MiddlewareFunc(handler rest.HandlerFunc) rest.HandlerFunc {
 
 			newLogger := oldLogger
 
-			newTraceRequest := app.NewID()
+			newTraceRequest := request.Header.Get(service.HTTPHeaderTraceRequest)
+			if newTraceRequest != "" {
+				if len(newTraceRequest) > _TraceMaximumLength {
```

Comment:
Now that a request can get the "trace request" id from an HTTP header it could be sent in from an outside caller. That means it is technically an attack vector. It goes along with the "validate all input" line of thinking. Under normal circumstances it will never be greater than the maximum length, but this is protecting from non-normal circumstances.